### PR TITLE
JWT support

### PR DESCRIPTION
**Pull request by jgr3go**
From #122 to support JWT

* Added support for [`django-rest-framework-jwt`](https://github.com/Tivix/django-rest-auth/compare/master...jgr3go:jwt-support?expand=1) instead of session tokens
* Enabled with `REST_USE_JWT`

Documentation probably needs some updates and I'd like the implementation approved, but submitting the pull now to start the discussion.

### CODE REVIEW

**Comment by auvipy** [comment]
@jgr3go  could you fix the merge conflict?

File: rest_auth/tests/requirements.pip [hunk 1]
```diff
@@ -1,3 +1,4 @@
 django-allauth>=0.19.1
 responses>=0.3.0
 flake8==2.4.0
+djangorestframework-jwt>=1.7.2
```
Comment (auvipy) [inline rest_auth/tests/requirements.pip]: no new line at the end

Comment (Akay7) [inline rest_auth/tests/requirements.pip]: why not, its not .py file and can be written without pip8

**Comment by gpbdev** [comment]
Would be great to see this get merged soon

**Comment by auvipy** [comment]
yeah

**Comment by jgr3go** [comment]
Oops, was on vacation and missed this message.  I'll try and fix the conflicts tonight.  I also need to add a bit to the documentation, and get tests created for this, but hopefully that shouldn't be too bad.

**Comment by jgr3go** [comment]
This looks good to go -- tests are passing, I've verified this works with my current production version. I think this should be ready to merge.

**Comment by auvipy** [comment]
@mariodev

**Comment by mariodev** [comment]
Thanks @jgr3go I'll do a quick review and merge asap.

**Comment by kibernick** [comment]
this is an awesome enhancement, would it be possible to merge soon?

**Comment by gpbdev** [comment]
Any updates on this? @mariodev

**Comment by tention** [comment]
Great enhancement, waiting for this.

**Comment by auvipy** [comment]
this branch has merge conflict

**Comment by Akay7** [comment]
0.7.0 already released, maybe change version in documentation?

**Comment by jgr3go** [comment]
Merge issues fixed. Anyone with privileges want to merge this and/or let me know whether there's anything preventing it from being merged?

**Comment by mariodev** [comment]
@jgr3go Looks like we're good to go now, sorry for making you wait so long. We had to postpone this merge until after releasing 0.7.

Thank you for putting all your work into it. This is a tremendous addition to DRA!

**Comment by cho-is** [comment]
Thank you for this!
I would appreciate some documentation.

I am testing it now but does not seem to return a jwt yet.

**Comment by cho-is** [comment]
I got it working using the Django REST framework JWT url but I wonder if it works with django rest auth LoginView?

**Comment by cho-is** [comment]
so I think pip package is not the latest

**Comment by jgr3go** [comment]
@cho-is it should work with LoginView if you have `REST_USE_JWT` enabled in your settings (and any appropriate settings for django-rest-framework-jwt  -- specifically, making sure `REST_FRAMEWORK` includes `'rest_framework_jwt.authentication.JSONWebTokenAuthentication'`, and `rest_framework_jwt` is included in your INSTALLED_APPS -- both pre-reqs for django-rest-framework-jwt which is why it's not in this documentation).

**Comment by jgr3go** [comment]
On second look, you're right. 0.8.0 hasn't been released yet. #185 to track that.

**Pull request merged**